addons/mail/models/discuss/discuss_channel_member.py

```diff
@@ -77,6 +78,38 @@ def _search_is_self(self, operator, operand):
                 ("guest_id", "!=", current_guest.id) if current_guest else expression.TRUE_LEAF,
             ]
 
+    def _search_is_pinned(self, operator, operand):
+        if (operator == "=" and operand) or (operator == "!=" and not operand):
+            return expression.OR(
+                [
+                    [("unpin_dt", "=", False)],
+                    [
+                        (
+                            "last_interest_dt",
+                            ">=",
+                            self._field_to_sql(self._table, "unpin_dt"),
+                        )
+                    ],
```

Comment:
would it be possible to single-line these parts of the domain into variables to reduce the size of this?

Reply:
i am not sure which is one is better to read, for me I think having this can be more readable? ( taking more lines but seems more clear )

Reply:
or something like this ?
```py
            unpin_dt_null = [("unpin_dt", "=", False)]
            last_interest_dt_no_later_than_unpin_dt = [
                ("last_interest_dt", ">=", self._field_to_sql(self._table, "unpin_dt"))
            ]
            channel_last_interest_dt_no_later_than_unpin_dt = [
                (
                    "channel_id.last_interest_dt",
                    ">=",
                    self._field_to_sql(self._table, "unpin_dt"),
                )
            ]
            return expression.OR(unpin_dt_null + last_interest_dt_no_later_than_unpin_dt + channel_last_interest_dt_no_later_than_unpin_dt)

```